src/Stratis.Features.FederatedPeg/CounterChainNetworkHolder.cs

```diff
@@ -0,0 +1,17 @@
+﻿using NBitcoin;
+
+namespace Stratis.Features.FederatedPeg
+{
+    /// <summary>
+    /// Holds information about the network we are connecting to.
+    /// </summary>
+    public class CounterChainNetworkHolder
```

Comment:
Why another class for it? 

I think you can set counter network at FederatedPegOptions and avoid adding counter network on DI.